src/raster/rasterband.rs

```diff
@@ -12,6 +15,68 @@ use ndarray::Array2;
 
 use crate::errors::*;
 
+/// Extra options used to read a raster.
+///
+/// For documentation, see `gdal_sys::GDALRasterIOExtraArg`.
+#[derive(Debug)]
+pub struct RasterIOExtraArg {
+    pub n_version: libc::c_int,
+    pub e_resample_alg: GDALRIOResampleAlg::Type,
+    pub pfn_progress: gdal_sys::GDALProgressFunc,
+    pub p_progress_data: *mut libc::c_void,
+    pub b_floating_point_window_validity: libc::c_int,
```

Comment:
can we change this to i32 or another integer from core?

Reply:
I guess an `usize` should be ok as well.

Just out of curiosity, what would be the difference? Just a "readability" thing? Or is there any technical reason?

Reply:
just readability